docs/DeveloperGuide.md

```diff
@@ -231,32 +236,59 @@ The following sequence diagram summarizes what happens when the user inputs an a
 
 ![AddGroupSequenceDiagram](images/AddGroupSequenceDiagram.png)
 
-### Add Allocation feature
+### Add Allocation : `addalloc`
+
+The `addalloc` feature allows users to allocate a student into a group.
+
+#### How the `addalloc` command works
+
+1. The user specifies the group name, the name or ID of the student to be allocated into the group.
```

Comment:
The user specifies the group name, and the name or ID of the student to be allocated into the group.